### PR TITLE
Add support for equipment stats to item stats plugin

### LINKED ISSUE #2519: Item comparison
Would be nice to have an item comparison to compare 2 items. Preferably how it was done in like 2009 or w/e it was made

### PR DESCRIPTION
**Pull request by deathbeam**
Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

Preview:
peek 2018-11-09 12-32

TODO:
- [x] Compare hovered item stats with currently equipped item
- [x] Add config option to disable either food or bonuses tooltips

Closes #2519

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java [hunk 1]
```diff
@@ -88,11 +93,78 @@ public Dimension render(Graphics2D graphics)
...
+
+			b.append(getChangeString("D Stab", e.getDstab(), false));
+			b.append(getChangeString("D Slash", e.getDslash(), false));
+			b.append(getChangeString("D Crush", e.getDcrush(), false));
+			b.append(getChangeString("D Magic", e.getDmagic(), false));
+			b.append(getChangeString("D Range", e.getDrange(), false));
+
+			b.append(getChangeString("Melee Str", e.getStr(), false));
+			b.append(getChangeString("Range Str", e.getRstr(), false));
+			b.append(getChangeString("Magic Dmg", e.getMdmg(), false));
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java]: This is a percentage so I feel like it should show a percentage sign in the tooltip.

**Comment by Nightfirecat** [comment]
I feel "D Crush" is a poor way to convey a defensive bonus, perhaps "Crush Def" would be a better concise way to write it?

**Comment by arthurtyukayev** [comment]
Why not just have it display the entire string unabbreviated?

**Comment by deathbeam** [comment]
Updated the naming with full versions @Nightfirecat @arthurtyukayev

**Comment by arthurtyukayev** [comment]
Now that I look at it, that's too many words. Maybe have it look like it does in the Equipment Screen in the game?

Bolded header and just the attack style?
image

**Comment by deathbeam** [comment]
I was thinking about headers but no, because it would look bad with items with little or no bonuses. Its either this or abbrews. @MagicfTail It is not finished yet as you would see if you would read discord :d

**Comment by MagicfTail** [comment]
Oops, must have missed the [wip] :p

**Comment by arthurtyukayev** [comment]
> I was thinking about headers but no, because it would look bad with items with little or no bonuses. Its either this or abbrevs. @MagicfTail It is not finished yet as you would see if you would read discord :d

Wouldn't the logic just be conditional? If there are any Defense stats it would show the header with the stats same with the Attack Stats? Can't you just not display the header if it had no stats?

**Comment by deathbeam** [comment]
Example:

```
Weight: +10
Attack Bonus
Slash: +10
Other Bonuses
Prayer: 4
Attack Speed: 4
``` that looks just horrible

**Comment by deathbeam** [comment]
Updated PR with new look, check OP for updated gif

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java (hunk 1 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java]: This is no longer an issue

**Comment by arthurtyukayev** [comment]
I like it.

**Comment by deathbeam** [comment]
Added support for comparison with currently equipped item, check OP for updated GIF

**Comment by deathbeam** [comment]
+ Added config options for enabling consumable/equipment stats, so this PR is no longer wip and is done. I am just dumping new item stats data locally with what I will update this PR after that finishes

**Comment by deathbeam** [comment]
Updated the PR to use https://static.runelite.net/cache/item/stats.json instead of file stored in client

**Pull request merged**